dkpro-core-api-io-asl/src/main/java/org/dkpro/core/api/io/JCasFileWriter_ImplBase.java

```diff
@@ -295,20 +294,22 @@ protected String getRelativePath(JCas aJCas)
             if (stripExtension) {
                 relativeDocumentPath = FilenameUtils.removeExtension(relativeDocumentPath);
             }
-
-            if (escapeDocumentId) {
-                try {
-                    relativeDocumentPath = URLEncoder.encode(relativeDocumentPath, "UTF-8");
-                }
-                catch (UnsupportedEncodingException e) {
-                    // UTF-8 must be supported on all Java platforms per specification. This should
-                    // not happen.
-                    throw new IllegalStateException(e);
-                }
+        }
+        
+        try {
+            if (escapeFilename) {
+                relativeDocumentPath = URLEncoder.encode(relativeDocumentPath, "UTF-8");
```

Comment:
I think we need the encode case only for the branch where documentID is used and the decode case only for the branch where baseURI/documentURI is used. If we leave it out here, then having `PARAM_ESCAPE_FILENAME` enabled would double-encode in the baseURI/documentURI case, right?